src/composeappmanager.cc

```diff
@@ -182,10 +192,18 @@ bool ComposeAppManager::fetchTarget(const Uptane::Target& target, Uptane::Fetche
   LOG_INFO << "Found " << cur_apps_to_fetch_and_update_.size() << " Apps to update";
 
   bool passed = true;
-  for (const auto& pair : cur_apps_to_fetch_and_update_) {
-    LOG_INFO << "Fetching " << pair.first << " -> " << pair.second;
-    if (!app_ctor_(pair.first).fetch(pair.second)) {
-      passed = false;
+  const auto& branch_to_pull = target.custom_data()["compose-apps-branch"].asString();
+  const auto& commit_hash_to_pull = target.custom_data()["compose-apps-hash"].asString();
+  if (!branch_to_pull.empty()) {
+    LOG_INFO << "Pulling from App Tree; branch: " << branch_to_pull << " commit: " << commit_hash_to_pull;
+    app_tree_.pull(config.ostree_server, keys, commit_hash_to_pull);
+    app_tree_.pull(config.ostree_server, keys, branch_to_pull);
```

Comment:
wouldn't pulling the commit hash be enough? what is pulling the branch doing?

Reply:
to update the branch/ref in the local ostree repo, otherwise, a user won't see the current/latest commit via `ostree --repo=<> refs`.